### PR TITLE
Support pip for curation

### PR DESCRIPTION
**Pull request by asafambar**
- [X] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [X] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [X] This pull request is on the dev branch.
- [X] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: utils/python/utils_test.go [hunk 1]
```diff
@@ -31,3 +33,32 @@ func initPoetryTest(t *testing.T) (string, func()) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			url, _, _, err := GetPypiRepoUrlWithCredentials(&config.ServerDetails{}, "test", tt.curationCmd)
+			require.NoError(t, err)
+			if tt.curationCmd {
+				assert.Contains(t, url.Path, "api/curation/audit")
+			} else {
+				assert.NotContains(t, url.Path, "api/curation/audit")
+
+			}
```
Comment (attiasas) [inline utils/python/utils_test.go:61]:
```suggestion
			if tt.curationCmd {
				assert.Contains(t, url.Path, "api/curation/audit/api/pypi/")
				assert.True(t, strings.HasSuffix(url.Path, "/simple")
			} else {
				assert.NotContains(t, url.Path, "api/curation/audit")

			}
```

**Review by attiasas — COMMENTED** [review]
Add description of the change to the PR

File: utils/python/utils.go [hunk 2]
```diff
@@ -41,6 +41,10 @@ func GetPypiRepoUrlWithCredentials(serverDetails *config.ServerDetails, reposito
 		}
 		password = serverDetails.GetAccessToken()
 	}
+	// In case of curation command, the download urls should be routed through a dedicated api.
+	if isCurationCmd {
+		rtUrl.Path += "api/curation/audit/"
+	}
 	rtUrl.Path += "api/pypi/" + repository + "/simple"
```
[the remark this answers is not in the document]
Comment (asafambar) [reply utils/python/utils.go:48]: yes

File: utils/python/utils_test.go [hunk 3]
```diff
@@ -31,3 +33,32 @@ func initPoetryTest(t *testing.T) (string, func()) {
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			url, _, _, err := GetPypiRepoUrlWithCredentials(&config.ServerDetails{}, "test", tt.curationCmd)
+			require.NoError(t, err)
+			if tt.curationCmd {
+				assert.Contains(t, url.Path, "api/curation/audit/api/pypi/")
+			} else {
+				assert.NotContains(t, url.Path, "api/curation/audit")
+
+			}
```
[the remark this answers is not in the document]
Comment (asafambar) [reply utils/python/utils_test.go]: Fixed

File: utils/python/utils.go [hunk 4]
```diff
@@ -41,6 +41,10 @@ func GetPypiRepoUrlWithCredentials(serverDetails *config.ServerDetails, reposito
 		}
 		password = serverDetails.GetAccessToken()
 	}
+	// In case of curation command, the download urls should be routed through a dedicated api.
+	if isCurationCmd {
+		rtUrl.Path += "api/curation/audit/"
```
[the remark this answers is not in the document]
Comment (asafambar) [reply utils/python/utils.go]: Fixed.

**Pull request merged**